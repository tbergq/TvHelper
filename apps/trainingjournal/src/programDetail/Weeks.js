// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from '@tbergq/tvhelper-relay';

import type { Weeks_program as ProgramType } from './__generated__/Weeks_program.graphql';

type Props = {|
  +program: ?ProgramType,
|};

export const Weeks = (props: Props) => {
  const weeks = props.program?.weeks?.edges ?? [];
  return (
    <div>
      {weeks.map(week => (
        <div key={week?.node?.id}>{week?.node?.name}</div>
      ))}
    </div>
  );
};

export default createFragmentContainer(Weeks, {
  program: graphql`
    fragment Weeks_program on Program {
      weeks {
        edges {
          node {
            id
            name
          }
        }
      }
    }
  `,
});
