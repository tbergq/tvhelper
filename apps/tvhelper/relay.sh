#!/bin/bash

cp ../graphql/schema.graphql ./schema.graphql
yarn relay-compiler --language typescript