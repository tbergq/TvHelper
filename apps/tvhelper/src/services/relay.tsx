import { ReactNode } from 'react';
import { Environment, Network, RecordSource, Store, FetchFunction } from 'relay-runtime';
import { RelayEnvironmentProvider } from 'react-relay';
import { RecordMap } from 'relay-runtime/lib/store/RelayStoreTypes';

interface Props {
  readonly children: ReactNode;
  readonly records?: RecordMap;
  readonly token: string | undefined;
}

interface CreateEnvironmentArgs {
  fetchQuery: FetchFunction;
  records?: RecordMap;
}
export const createEnvironment = ({ fetchQuery, records }: CreateEnvironmentArgs) => {
  const network = Network.create(fetchQuery);

  const store = new Store(new RecordSource(records));

  return new Environment({
    network,
    store,
  });
};

const { GRAPHQL_URL } = process.env;

const getBody = (operation, variables) => {
  if (operation.id) {
    return {
      queryId: operation.id,
      variables,
    };
  }
  return {
    query: operation.text,
    variables,
  };
};

export const makeFetchQuery = (token, baseUrl = '') => {
  const fetchQuery = async (operation, variables /* , cacheConfig, uploadables */) => {
    const res = await fetch(`${baseUrl}`, {
      method: 'POST',
      headers: {
        // Add authentication and other headers here
        'content-type': 'application/json',
        'authorization': token,
      },
      body: JSON.stringify(getBody(operation, variables)),
    });
    return res.json();
  };
  return fetchQuery;
};

const EnvironmentProvider = ({ children, records, token }: Props) => {
  const environment = createEnvironment({
    fetchQuery: makeFetchQuery(token, GRAPHQL_URL),
    records,
  });

  return <RelayEnvironmentProvider environment={environment}>{children}</RelayEnvironmentProvider>;
};

export { EnvironmentProvider };
