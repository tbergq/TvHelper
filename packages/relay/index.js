// @flow

export { default as QueryRenderer } from './src/QueryRenderer';
export { graphql, createFragmentContainer } from '@kiwicom/relay';
