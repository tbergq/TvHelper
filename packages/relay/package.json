{
  "name": "@tbergq/relay",
  "private": true,
  "main": "index.js",
  "version": "0.0.1",
  "sideEffects": false,
  "dependencies": {
    "@adeira/fetch": "^1.0.2",
    "@adeira/js": "^1.2.1",
    "@adeira/relay": "^1.3.0",
    "@adeira/relay-utils": "^0.7.0",
    "@tbergq/components": "*",
    "graphql": "^15.0.0",
    "react-fast-compare": "^3.1.1"
  },
  "peerDependencies": {
    "react": "",
    "react-relay": "*",
    "relay-runtime": ""
  },
  "devDependencies": {
    "relay-test-utils": "^9.1.0"
  }
}
