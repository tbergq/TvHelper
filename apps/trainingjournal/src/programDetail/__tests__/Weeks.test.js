// @flow

import * as React from 'react';
import { QueryRenderer, graphql } from '@kiwicom/relay';
import { createMockEnvironment, MockPayloadGenerator } from 'relay-test-utils';
import { create } from 'react-test-renderer';

import Weeks from '../Weeks';

const renderInner = props => <Weeks program={props.program} />;

test('Weeks', () => {
  const environment = createMockEnvironment();
  const TestRenderer = () => (
    <QueryRenderer
      environment={environment}
      query={graphql`
        query WeeksQuery @relay_test_operation {
          program(programId: "123") {
            ...Weeks_program
          }
        }
      `}
      variables={{}}
      onResponse={renderInner}
    />
  );
  const renderer = create(<TestRenderer />);
  environment.mock.resolveMostRecentOperation(operation =>
    MockPayloadGenerator.generate(operation),
  );

  expect(renderer).toMatchSnapshot();
});
