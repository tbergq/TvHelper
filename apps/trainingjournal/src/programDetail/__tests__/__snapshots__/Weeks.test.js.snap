// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weeks 1`] = `
.c9 {
  width: 24px;
  height: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
  color: #5F738C;
}

.c14 {
  font-family: "Roboto",-apple-system,".SFNSText-Regular","San Francisco","Segoe UI","Helvetica Neue","Lucida Grande",sans-serif;
  font-size: 16px;
  font-weight: 500;
  color: #252A31;
  line-height: 24px;
  margin: 0;
}

.c8 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-transition: 0.15s;
  transition: 0.15s;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: margin 0.15s linear;
  transition: margin 0.15s linear;
  cursor: pointer;
  position: relative;
  min-height: 24px;
  margin: -16px;
  padding: 16px;
}

.c5:hover {
  background: #F5F7F9;
}

.c5 .c7 {
  margin-left: 16px;
}

.c5:focus {
  background: #F5F7F9;
  outline: none;
}

.c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c11 {
  position: relative;
  -webkit-transition: max-height 0.15s linear;
  transition: max-height 0.15s linear;
  max-height: 0px;
  overflow: hidden;
  visibility: hidden;
}

.c10 {
  font-family: "Roboto",-apple-system,".SFNSText-Regular","San Francisco","Segoe UI","Helvetica Neue","Lucida Grande",sans-serif;
  font-size: 14px;
  line-height: 1.4;
  color: #252A31;
  border-top: 0px solid #EFF2F5;
  -webkit-transition: padding 0.15s linear,border-top 0.15s linear;
  transition: padding 0.15s linear,border-top 0.15s linear;
}

.c4 {
  width: 100%;
  padding: 16px;
  box-sizing: border-box;
  position: relative;
  background: #FFFFFF;
}

.c2 {
  -webkit-transition: margin 0.15s ease-in-out;
  transition: margin 0.15s ease-in-out;
}

.c2 .c3,
.c2 .c16,
.c2 > .Loading__StyledLoading-sc-1psg3na-0 {
  border-left: 1px solid #EFF2F5;
  border-right: 1px solid #EFF2F5;
  border-bottom: 1px solid #EFF2F5;
  background: #FFFFFF;
}

.c0 {
  width: 100%;
  box-sizing: border-box;
  position: relative;
  font-family: "Roboto",-apple-system,".SFNSText-Regular","San Francisco","Segoe UI","Helvetica Neue","Lucida Grande",sans-serif;
}

.c0 .c1:first-of-type .c16,
.c0 .c1:first-of-type .c3,
.c0 .c1:first-of-type > .Loading__StyledLoading-sc-1psg3na-0 {
  border-top: 1px solid #EFF2F5;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
}

.c0 .c1:last-of-type .c16,
.c0 .c1:last-of-type .c3 {
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
}

.c13 {
  margin-right: 8px;
  margin-bottom: 8px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c15 {
  margin-bottom: 8px;
}

@media (min-width:992px) {
  .c5 {
    padding: 24px;
    margin: -24px;
  }
}

@media (min-width:992px) {
  .c4 {
    padding: 24px;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1 c2"
    >
      <div
        className="c3 c4"
      >
        <div
          aria-expanded={false}
          className="c5"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="c6"
          >
            &lt;mock-value-for-field-"name"&gt;
          </div>
          <svg
            className="c7 c8 c9"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
        <div
          className="c10"
        >
          <div
            aria-hidden={true}
            className="c11"
          >
            <div>
              <div>
                <div
                  className="c12"
                >
                  <div
                    className="c13"
                  >
                    <h3
                      className="c14"
                    >
                      <a
                        href="/programs/day?dayId=<Day-mock-id-2>"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        &lt;mock-value-for-field-"name"&gt;
                      </a>
                    </h3>
                    <div>
                      &lt;mock-value-for-field-"name"&gt;
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c15"
  />
</div>
`;
